src/test/java/org/hobsoft/spring/resttemplatelogger/LoggingInterceptorTest.java

```diff
@@ -63,61 +59,37 @@ public void setUp()
 	{
 		when(log.isDebugEnabled()).thenReturn(true);
 		
-		loggingInterceptor = new LoggingInterceptor(log);
+		loggingInterceptor = new LoggingInterceptor(log, formatter);
 	}
 	
 	// ----------------------------------------------------------------------------------------------------------------
 	// tests
 	// ----------------------------------------------------------------------------------------------------------------
 	
 	@Test
-	public void canLogRequestUsingDefaultCharset() throws IOException, URISyntaxException
-	{
-		MockClientHttpRequest request = new MockClientHttpRequest(POST, new URI("/hello"));
-		byte[] body = "world £".getBytes(ISO_8859_1);
-		when(execution.execute(request, body)).thenReturn(new MockClientHttpResponse(new byte[0], OK));
-		
-		loggingInterceptor.intercept(request, body, execution);
-		
-		verify(log).debug("Request: POST /hello world £");
-	}
-	
-	@Test
-	public void canLogRequestUsingCharset() throws IOException, URISyntaxException
+	public void canLogRequest() throws IOException
 	{
-		MockClientHttpRequest request = new MockClientHttpRequest(POST, new URI("/hello"));
-		request.getHeaders().setContentType(parseMediaType("text/plain;charset=UTF-8"));
-		byte[] body = "world £".getBytes(UTF_8);
-		when(execution.execute(request, body)).thenReturn(new MockClientHttpResponse(new byte[0], OK));
+		MockClientHttpRequest request = new MockClientHttpRequest();
+		byte[] body = new byte[] {(byte) 1};
+		when(formatter.formatRequest(request, body)).thenReturn("message");
 		
 		loggingInterceptor.intercept(request, body, execution);
 		
-		verify(log).debug("Request: POST /hello world £");
+		verify(log).debug("message");
 	}
 	
 	@Test
-	public void canLogResponseUsingDefaultCharset() throws IOException, URISyntaxException
+	public void canLogResponse() throws IOException
 	{
 		MockClientHttpRequest request = new MockClientHttpRequest();
-		byte[] body = "hello £".getBytes(ISO_8859_1);
-		when(execution.execute(request, new byte[0])).thenReturn(new MockClientHttpResponse(body, OK));
+		byte[] body = new byte[] {(byte) 1};
 		
-		loggingInterceptor.intercept(request, new byte[0], execution);
+		ClientHttpResponse response = withSuccess().createResponse(null);
```

Comment:
Just curious, why use `MockRestResponseCreators` rather than:
```
MockClientHttpResponse response = new MockClientHttpResponse(body, OK);
```

Reply:
Small, clearly unnecessary thing but... I didn't want to use `MockClientHttpResponse` if I could avoid it because it takes a `body`. I didn't want that to get confused with the other `body`s that actually are pertinent to the test. So I wondered if there was a factory for responses that didn't take `body`s. Which, I was pleased to see, there was, so I used it. Whether this is actually any clearer than just `new byte[0]`, I'll let you decide... :)